server/controllers/LikeController.js

```diff
@@ -0,0 +1,178 @@
+import db from '../database/models';
+import helper from '../utils/helpers/likeControllerHelper';
+
+const { Article, ArticleLike, User } = db;
+
+/**
+ * This class contains static methods for liking and unliking
+ * articles in the applications. It also contains methods
+ * for retrieving like informations for a particular article
+ * @class LikeController
+ * @returns {object} when called with the  new it returns an
+ * instance of LikeController but this is never used
+ */
+class LikeController {
+  /**
+   * This function is a middleware and is called when the user wants
+   * to like a particular article.
+   * @param {*} req the request object containing the user's request
+   * @param {*} res the response object used to send a respons to the user
+   * @param {*} next called when an error occurs
+   * @returns {void} does not return a value
+   * @author Chidiebere Oguejiofor
+   */
+  static likeArticle(req, res, next) {
+    const { slug } = req.params;
+    const { decoded: user } = req.body;
+    Article.findOne({
+      where: {
+        slug,
+      },
+      include: [
+        {
+          model: User,
+          as: 'author',
+          attributes: [
+            'firstName', 'lastName',
+            'username', 'image', 'bio',
+            'id',
+          ],
+        },
+      ],
+      attributes: ['id', 'title', 'slug'],
+    }).then((article) => {
+      if (article) {
+        const likeObj = {
+          userId: user.id,
+          articleId: article.dataValues.id,
```

Comment:
You can use 'article.addArticleLikes' since you have already established associations to reduce the number of lines of code and avoid specifying findOrcreate with the where clause